### ISSUE TITLE
Can't connect to Linux Hyper-V Machines, RDCman works fine

### ISSUE THREAD

**Issue by davi1td** [mRemoteNG/mRemoteNG#1673]
VMRDP Connections do not work when connecting to Linux Guest VM's.
Can't connect to Debian 10 , Latest Hyper-v supported version per : [Debian-Supported-Hyper-v](https://docs.microsoft.com/en-us/windows-server/virtualization/hyper-v/supported-debian-virtual-machines-on-hyper-v)

## Expected Behavior
Successfully connect to guest Linux OS. 

## Current Behavior
Receive the following generic error when trying to connect to a Linux host:
Protocol Event Disconnected. Host: "localhost"; Protocol: "RDP" Message: "An internal error has occurred."

## Steps to Reproduce (for bugs)
Simply create a connection & Enable Hyper-v as per [#1422](https://github.com/mRemoteNG/mRemoteNG/pull/1422)
Pointing to a Linux host. 
Hostname/IP: LocalHost
Port: 2179
Use VIM ID: Yes
VIM ID: xxxxxxxx-xxxxxx

## Context
This is purely a connection issue, RDCman works fine for connecting to Linux hosts using VMRDP mode. 

## Your Environment
Debian 10
mRemoteNG Version used: 1.77.1.27654
Microsoft Windows [Version 10.0.18362.592] x64

**Comment by farosch** [comment]
can you connect to windows hyper-v machines?
have you tried 127.0.0.1 instead of localhost?

**Comment by Shananiganeer** [comment]
I get the same internal error when attempting to connect to a Windows Hyper-V host. Tried 127.0.0.1, localhost, hostname, and LAN IP.

**Comment by Sascha-69** [comment]
Have you looked at [Connect to virtual machine on Hyper-V](https://github.com/mRemoteNG/mRemoteNG/blob/develop/mRemoteNG/Documentation/howtos/vmrdp.rst)

**Comment by davi1td** [comment]
Just tried:
----------
WarningMsg
08/27/2020 14:45:24
Protocol Event Disconnected. Host: "127.0.0.1"; Protocol: "RDP" Message: "An internal error has occurred."
----------

**Comment by Shananiganeer** [comment]
> 
> 
> Have you looked at [Connect to virtual machine on Hyper-V](https://github.com/mRemoteNG/mRemoteNG/blob/develop/mRemoteNG/Documentation/howtos/vmrdp.rst)

Yes I read through the docs, this feature seemingly doesn't work.

**Comment by Shananiganeer** [comment]
Don't know what changed, but had to reboot for updates and then this worked for the first time today.

**Comment by davi1td** [comment]
> 
> 
> Don't know what changed, but had to reboot for updates and then this worked for the first time today. Made no config changes.

Not for me, i'm running pre-release 1.77.1.27654, what I don't get is it's from Sept 2019 and is latest dev release??
Anyway what version are you running?

**Comment by Shananiganeer** [comment]
I'm running the same version that you are.

**Comment by davi1td** [comment]
Cool thanks for update, clearly this isn't stable at all. I just don't understand the dev/release process, this version is dev/beta and over a year old ...

**Comment by davi1td** [comment]
OK, Update got it working using the mRemoteNG-removecitrix(1.77.1.23492)., complied and works with localhost or 127.0.0.1. However, even though enhanced mode is enabled in hyper-v it will not work when "use enhanced mode" is enabled in mRemoteNG.
Exact same settings in 1.77.127654, except for "use enhanced mode" which doesn't exist, so what would that indicate?

Shananiganeer, do you have enhanced mode enabled in Hyper-v ? Hyper-V settings => Enhanced Session Mode Policy Allow Enhanced Session Mode

**Comment by Shananiganeer** [comment]
I have Enhanced Session Mode enabled in Hyper-V, but have no corresponding setting within mRemoteNG.

**Comment by davi1td** [comment]
YA, that setting is in the version i mentioned i compiled. 
I tried turning enhanced mode off in Hyper-V with no luck still 1.77.1.27654, but at least it appears to be working some day in the future lol, good you got it working who knows why.

**Comment by Shananiganeer** [comment]
Here's my config, though I doubt you've got anything wrong.

![image](https://user-images.githubusercontent.com/20528114/91751742-4c3ce300-eb93-11ea-8b32-232809d95165.png)

**Comment by davi1td** [comment]
Yup, thanks mine are the same.

**Comment by davi1td** [comment]
Like I said it does work in the latest build i compiled, which has the use enhanced and only works for me when disabled, thats why I was curious if you had it enabled in hyper-v, but I've messed with that as well with no luck. 
I'll just create the installer for the one I got working and use it instead ...
